Cubical/Codata/M/AsLimit/stream.agda

```diff
@@ -24,7 +29,167 @@ cons : ∀ {A} -> A -> stream A -> stream A
+        (snd (tl (Stream-to-stream b)) n)
+        (Stream-to-stream-func-π n (tail b))
+
+tl-to-tail :
+  ∀ {A : Set} (b : Stream A)
+  → tl (Stream-to-stream b) ≡ Stream-to-stream (tail b)
+tl-to-tail {A = A} b =
+  ΣPathP (funExt (tail-eq-x b) , λ i n j → tail-eq-π b n i j)
+
+nth : ∀ {A : Set} → ℕ → (b : Stream A) → A
+nth 0 b = head b
+nth (suc n) b = nth n (tail b)
+
+stream-equality-iso-1 : ∀ {A : Set} → (b : Stream A) → stream-to-Stream (Stream-to-stream b) ≡ b
+stream-equality-iso-1 b = bisim-nat (stream-to-Stream (Stream-to-stream b)) b (helper b)
+  where
+    helper : ∀ {A : Set} → (b : Stream A) → ((n : ℕ) → nth n (stream-to-Stream (Stream-to-stream b)) ≡ nth n b)
+    helper b 0 = head-to-hd (Stream-to-stream b) ∙ hd-to-head b
+    helper b (suc n) =
+      nth (suc n) (stream-to-Stream (Stream-to-stream b))
+        ≡⟨ refl ⟩
+      nth n (tail (stream-to-Stream (Stream-to-stream b)))
+        ≡⟨ cong (nth n) (tail-to-tl (Stream-to-stream b) ∙ cong stream-to-Stream (tl-to-tail b)) ⟩
+      nth n (stream-to-Stream (Stream-to-stream (tail b)))
+        ≡⟨ helper (tail b) n ⟩
+      nth n (tail b) ∎
+
+    bisim-nat : ∀ {A : Set} → (a b : Stream A) → ((n : ℕ) → nth n a ≡ nth n b) -> a ≡ b
+    bisim-nat a b nat-bisim = bisim (bisim-nat' a b nat-bisim)
+      where
+        open Equality≅Bisimulation
+        open _≈_
+
+        bisim-nat' : ∀ {A : Set} → (a b : Stream A) → ((n : ℕ) → nth n a ≡ nth n b) -> a ≈ b
+        ≈head (bisim-nat' a b nat-bisim) = nat-bisim 0
+        ≈tail (bisim-nat' a b nat-bisim) = bisim-nat' (tail a) (tail b) (nat-bisim ∘ suc)
+
+private
+  stream-equality-iso-2-x : ∀ {A : Set} → (a : stream A) (n : ℕ) → Stream-to-stream-func-x n (stream-to-Stream a) ≡ fst a n
+  stream-equality-iso-2-x a 0 = refl
+  stream-equality-iso-2-x {A} a (suc n) =
+    Stream-to-stream-func-x (suc n) (stream-to-Stream a)
+      ≡⟨ refl ⟩
+    (head (stream-to-Stream a) , (λ _ → Stream-to-stream-func-x n (tail (stream-to-Stream a))))
+      ≡⟨ ΣPathP (head-to-hd a , funExt λ _ → cong (Stream-to-stream-func-x n) (tail-to-tl a)) ⟩
+    (hd a , (λ _ → Stream-to-stream-func-x n (stream-to-Stream (tl a))))
+      ≡⟨ ΣPathP (refl , funExt λ _ → stream-equality-iso-2-x (tl a) n) ⟩
+    (hd a , (λ _ → fst (tl a) n))
+      ≡⟨ ΣPathP (refl , refl) ⟩
+    (out-fun a .fst , (λ _ → fst (tl a) n))
+      ≡⟨ ΣPathP (refl , refl) ⟩
+    (fst (fst a (suc 0)) , (λ _ → fst (tl a) n))
+      ≡⟨ ΣPathP (temp n , refl) ⟩
+    (fst (fst a (suc n)) , (λ _ → fst (tl a) n))
+      ≡⟨ ΣPathP (refl , funExt λ _ → temp') ⟩
+    fst (fst a (suc n)) , snd (fst a (suc n))
+      ≡⟨ refl ⟩
+    fst a (suc n) ∎
+    where
+      temp : ∀ (n : ℕ) → fst (fst a (suc 0)) ≡ fst (fst a (suc n))
+      temp 0 = refl
+      temp (suc n) = temp n ∙ cong fst (sym (snd a (suc n)))
+
+      temp' : fst (tl a) n ≡ snd (fst a (suc n)) tt
+      temp' = sym (transport-filler (λ i → Wₙ (stream-S A) n) (a .fst (suc n) .snd tt))
+
+  postulate
```

Comment:
Why postulate?

Reply:
I have not been able to complete the proof.